tools/sof-tplgreader.py

```diff
@@ -16,43 +16,40 @@ def __init__(self):
     def __comp_pipeline(self, pipeline):
         return pipeline['id']
 
-    # fork & split from TplgFormatter
-    def _fmt2str(self, cap):
-        fmt = cap['formats']
-        if fmt & (1 << 2) != 0:
-            return "S16_LE"
-        elif fmt & (1 << 6) != 0:
-            return "S24_LE"
-        elif fmt & (1 << 10) != 0:
-            return "S32_LE"
-        elif fmt & (1 << 14) != 0:
-            return "FLOAT"
-        else: # without match
-            return ""
+    # supported formats of playback pipeline in formats[0]
+    # supported formats of capture pipeline in formats[1]
+    def _fmt2str(self, formats, pcm_type):
```

Comment:
for this tools, don't need get full list, so you can direct to access target list